Realm/RLMArrayLinkView.mm

```diff
@@ -194,15 +200,11 @@ - (void)replaceObjectAtIndex:(NSUInteger)index withObject:(RLMObject *)object {
 - (NSUInteger)indexOfObject:(RLMObject *)object {
     // check attached for table and object
     RLMLinkViewArrayValidateAttached(self);
+
     if (object->_realm && !object->_row.is_attached()) {
         @throw RLMException(@"RLMObject is no longer valid");
     }
 
-    // check that object types align
```

Comment:
why was this removed? it should be a programmer error if you're passing in objects that can't be contained by the RLMArray.